src/Pomelo.EntityFrameworkCore.MySql/Storage/Internal/MySqlTypeMapper.cs

```diff
@@ -60,9 +68,27 @@ public class MySqlTypeMapper : RelationalTypeMapper
 
         readonly Dictionary<string, RelationalTypeMapping> _simpleNameMappings;
         readonly Dictionary<Type, RelationalTypeMapping> _simpleMappings;
-
-        public MySqlTypeMapper()
+        
+        public MySqlTypeMapper(IServiceProvider services)
         {
+            this.services = services;
+            var relationalConn = services.GetService<MySqlRelationalConnection>();
+            var TreatTinyIntAsBoolean = true;
+            var UseDateTime6 = false;
+            if (relationalConn != null)
+            {
+                var csb = new MySqlConnectionStringBuilder(relationalConn.DbConnection.ConnectionString);
+                TreatTinyIntAsBoolean = csb.TreatTinyAsBoolean;
+                relationalConn.DbConnection.Open();
+                using (var cmd = new MySqlCommand("SELECT VERSION();", relationalConn.DbConnection as MySqlConnection))
```

Comment:
Oh, if TypeMapper is only loaded in Migrations and Scaffolding then it should be OK to open a connection here.  As long as it does not get loaded during normal operation.